engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/EngineJobHashingActor.scala

```diff
@@ -127,41 +208,56 @@ object EngineJobHashingActor {
 
 /**
   * Transient data for the EJHA.
   *
   * @param possibleCacheResults The set of cache results which have matched all currently tried hashes
-  * @param keysCheckedAgainstCache The set of hash keys which have already been compared against the cache result table
+  * @param remainingCacheChecks The set of hash keys which have not yet had their cache results fetched
   * @param hashesKnown The set of all hashes calculated so far (including initial hashes)
-  * @param hashesNeeded Not transient but oh-so-useful for everything else.
+  * @param remainingHashesNeeded The set of hashes which are still needed for writing to the database
   */
-private[callcaching] case class EJHAData(possibleCacheResults: Set[MetaInfoId],
-                                         keysCheckedAgainstCache: Set[HashKey],
+private[callcaching] case class EJHAData(possibleCacheResults: Option[Set[MetaInfoId]],
+                                         remainingCacheChecks: Set[HashKey],
                                          hashesKnown: Set[HashResult],
-                                         hashesNeeded: Set[HashKey]) {
-
+                                         remainingHashesNeeded: Set[HashKey]) {
   // Manipulators
   def intersectCacheResults(newCacheResults: CacheResultMatchesForHashes) = {
+    System.out.println(s"Intersecting new hashkeys: ${newCacheResults.hashResults.map(_.hashKey.key).mkString(", ")}")
     val newIds = newCacheResults.cacheResultIds
-    val intersectedIds = if (keysCheckedAgainstCache.nonEmpty) possibleCacheResults.intersect(newIds) else newIds
+    val intersectedIds = possibleCacheResults match {
+      case None => newIds
+      case Some(currentCacheResults) => currentCacheResults.intersect(newIds)
+    }
+    this.copy(
+      possibleCacheResults = Option(intersectedIds),
+      remainingCacheChecks = remainingCacheChecks.diff(newCacheResults.hashResults.map(_.hashKey))
+    )
+  }
+  def withNewKnownHashes(hashResults: Set[HashResult]) = {
     this.copy(
-      possibleCacheResults = intersectedIds,
-      keysCheckedAgainstCache = keysCheckedAgainstCache ++ newCacheResults.hashResults.map(_.hashKey))
+      hashesKnown = hashesKnown ++ hashResults,
+      remainingHashesNeeded = remainingHashesNeeded.diff(hashResults.map(_.hashKey)))
   }
-  def withNewKnownHashes(hashResults: Set[HashResult]) = this.copy(hashesKnown = hashesKnown ++ hashResults) // NB doesn't update hashesNeeded which should be considered immutable
 
   // Queries
-  def allHashesKnown = hashesNeeded.size == hashesKnown.size
-  def allCacheResultsIntersected = keysCheckedAgainstCache.size == hashesNeeded.size
-  def cacheHit = if (allCacheResultsIntersected && possibleCacheResults.nonEmpty) possibleCacheResults.headOption else None
+  def allHashesKnown = remainingHashesNeeded.isEmpty
+  def allCacheResultsIntersected = remainingCacheChecks.isEmpty
+  def cacheHit = if (allCacheResultsIntersected) possibleCacheResults.flatMap(_.headOption) else None
```

Comment:
More general statement across the whole PR to prefer the HOF style preferred by the scala style guide